### ISSUE TITLE
Grads w.r.t. weights of `MixtureGeneral` Distribution are giving `nan`s

### ISSUE THREAD

**Issue by Qazalbash** [pyro-ppl/numpyro#1870]
Hi,

We have created some models where we estimate the weights of the `MixtureGeneral` distribution. However, when computing the gradient of this argument, we are encountering `nan` values. We enabled `jax.config.update("debug_nan", True)` to diagnose the issue, and it pointed to the following line:

https://github.com/pyro-ppl/numpyro/blob/8e9313fd64a34162bc1c08b20ed310373e82e347/numpyro/distributions/mixtures.py#L152

I suspect that after the implementation of https://github.com/pyro-ppl/numpyro/pull/1791, extra care is needed to handle `inf` and `nan` values, possibly by using a double `where` for a safe `logsumexp`.

> [!IMPORTANT]
> This is an urgent issue, so a prompt response would be greatly appreciated.

**Comment by fehiepsi** [comment]
You can add `jax.debug.print(...)` to inspect the component log probs. If all of the component log probs are -inf, nan will happen.

**Comment by Qazalbash** [comment]
So I checked and I got no `nan`s but some `inf`s not all.

Let me explain what I am trying to do, suppose we have a mixture model with $ p_i(x|\Lambda) $ as the probability of each distribution. We also have $ \mathcal{R}_i $ as the scaling factor of each component, they are usually greater than 1. To make them work with `MixtureGeneral` we normalize them and pass it to the `CategoricalProb` as a mixing distribution.

So the compute happens like,

$$
\log(p(x|\Lambda))=\sum_{i=1}^{n}\mathcal{R}_ip_i(x|\Lambda) \iff \log(p(x|\Lambda))=\sum_{j=1}^{n}R_j+\sum_{i=1}^{n}\frac{\mathcal{R}_i}{\sum_{j=1}^{n}R_j}p_i(x|\Lambda)
$$

We are trying to estimate the $\log(\mathcal{R}_i)$ along with some other parameters. We do it like,

```python
model = ... # Pass parameters to initialize the MixtureGeneral object
log_rate = ... # array of log of rates
log_sum_of_rates = jax.nn.logsumexp(log_rates, axis=-1) # log(R_0 + R_1 + ... + R_n)
mixing_probs = jax.nn.softmax(log_rates, axis=-1) # Normalized rates [R_0 / sum(R_i), R_1 / sum(R_i), ..., R_n / sum(R_i)]
model._mixing_distribution = CategoricalProbs(probs=mixing_probs)
log_p = model.log_prob(y) # Calculate probability
log_p = log_p + log_sum_of_rates # adding back the extra we subtracted to normalize
```
